pkg/controller/master/backup/backup_status.go

```diff
@@ -28,17 +49,34 @@ func (h *Handler) updateConditions(vmBackup *harvesterv1.VirtualMachineBackup) e
 
 	ready := true
 	errorMessage := ""
-	for _, vb := range vmBackup.Status.VolumeBackups {
+	var volumeSizeSum int64
+	var progressWeightSum int64
+
+	for i := range vmBackupCpy.Status.VolumeBackups {
+		vb := &vmBackupCpy.Status.VolumeBackups[i]
 		if vb.ReadyToUse == nil || !*vb.ReadyToUse {
 			ready = false
 		}
 
+		if vmBackupCpy.Spec.Type == harvesterv1.Backup {
+			if err := h.updateBackupProgress(vb); err != nil {
+				return err
+			}
+
+			volumeSizeSum += vb.VolumeSize
+			progressWeightSum += int64(vb.BackupProgress) * vb.VolumeSize
+		}
+
 		if vb.Error != nil {
 			errorMessage = fmt.Sprintf("VolumeSnapshot %s in error state", *vb.Name)
 			break
 		}
 	}
 
+	if volumeSizeSum != 0 {
+		vmBackupCpy.Status.Progress = int(progressWeightSum / volumeSizeSum)
```

Comment:
I don't think it matters in practice, so feel free to ignore, but you could add 0.5 before casting to an int in order to round to the nearest integer. It gives a slightly more accurate representation since without it, 11.8% would be displayed as 11% but most people would probably be fine approximating 11.8% as 12%.

```suggestion
		vmBackupCpy.Status.Progress = int((progressWeightSum / volumeSizeSum)+0.5)
```

Reply:
I'd like to keep it, I think reporting less progress may be better than over-reporting, thanks.